engine/src/main/java/com/datatorrent/stram/codec/LogicalPlanSerializer.java

```diff
@@ -116,91 +204,96 @@ public boolean useForType(JavaType t)
-      String sourcePortName = streamMeta.getSource().getPortName();
-      OperatorMeta operatorMeta = streamMeta.getSource().getOperatorMeta();
-      HashMap<String, Object> sourcePortDetailMap = new HashMap<String, Object>();
-      sourcePortDetailMap.put("operatorName", operatorMeta.getName());
-      sourcePortDetailMap.put("portName", sourcePortName);
-      streamDetailMap.put("name", streamName);
-      streamDetailMap.put("source", sourcePortDetailMap);
-      List<InputPortMeta> sinks = streamMeta.getSinks();
-      ArrayList<HashMap<String, Object>> sinkPortList = new ArrayList<HashMap<String, Object>>();
-      for (InputPortMeta sinkPort : sinks) {
-        HashMap<String, Object> sinkPortDetailMap = new HashMap<String, Object>();
-        sinkPortDetailMap.put("operatorName", sinkPort.getOperatorWrapper().getName());
-        sinkPortDetailMap.put("portName", sinkPort.getPortName());
-        sinkPortList.add(sinkPortDetailMap);
+  private static Object getLogicalStreamDetails(StreamMeta streamMeta)
+  {
+
+    HashMap<String, Object> streamDetailMap = new HashMap<String, Object>();
+    String streamName = streamMeta.getName();
+    String sourcePortName = streamMeta.getSource().getPortName();
+    OperatorMeta operatorMeta = streamMeta.getSource().getOperatorMeta();
+    HashMap<String, Object> sourcePortDetailMap = new HashMap<String, Object>();
+    sourcePortDetailMap.put("operatorName", operatorMeta.getName());
+    sourcePortDetailMap.put("portName", sourcePortName);
+    streamDetailMap.put("name", streamName);
+    streamDetailMap.put("source", sourcePortDetailMap);
+    List<InputPortMeta> sinks = streamMeta.getSinks();
+    ArrayList<HashMap<String, Object>> sinkPortList = new ArrayList<HashMap<String, Object>>();
+    for (InputPortMeta sinkPort : sinks) {
+      HashMap<String, Object> sinkPortDetailMap = new HashMap<String, Object>();
+      sinkPortDetailMap.put("operatorName", sinkPort.getOperatorWrapper().getName());
+      sinkPortDetailMap.put("portName", sinkPort.getPortName());
+      sinkPortList.add(sinkPortDetailMap);
+    }
+    streamDetailMap.put("sinks", sinkPortList);
+    if (streamMeta.getLocality() != null) {
+      streamDetailMap.put("locality", streamMeta.getLocality().name());
+    }
+    return streamDetailMap;
+  }
+
+  private static Map<String, Object> fillLogicalModuleDetails(ModuleMeta moduleMeta, LogicalPlan dag, boolean flatten)
+  {
+    Map<String, Object> moduleDetailMap = new HashMap<String, Object>();
+    ArrayList<Object> operatorArray = new ArrayList<Object>();
+    ArrayList<Object> streamArray = new ArrayList<Object>();
+    moduleDetailMap.put("name", moduleMeta.getName());
+    moduleDetailMap.put("className", moduleMeta.getModule().getClass().getName());
+    moduleDetailMap.put("operators", operatorArray);
+    moduleDetailMap.put("streams", streamArray);
+    if (flatten) {
+      for (OperatorMeta operatorMeta : dag.getAllOperators()) {
```

Comment:
why iterate over all the operators.. Can it not be `moduleMeta.getDag().getAllOperators()`?